azure/scope/machine.go

```diff
@@ -168,10 +133,14 @@ func (m *MachineScope) VMSpec() azure.ResourceSpecGetter {
 		AdditionalCapabilities: m.AzureMachine.Spec.AdditionalCapabilities,
 		ProviderID:             m.ProviderID(),
 	}
-	if m.cache != nil {
-		spec.SKU = m.cache.VMSKU
-		spec.Image = m.cache.VMImage
-		spec.BootstrapData = m.cache.BootstrapData
+	if bootstrapData, err := m.GetBootstrapData(ctx); err == nil {
```

Comment:
We should be handling the errors here and below

if I remember correctly that was one of the main reasons we decided to go with a separate `InitMachineCache` func which would be able to handle errors and pass in context without requiring those on the Spec functions.